### PR TITLE
[linux] Make BLE connection retry on SW connection abort error

### PR DESCRIPTION
**Pull request by LuDuda**
#### Problem
Due to radio interferences or Wi-Fi coexistence, sometimes the BLE connection may not be
established (e.g. Connection Indication Packet is missed by BLE peripheral). In such case,
chip-tool returns `FAIL: ConnectDevice : GDBus.Error:org.bluez.Error.Failed: Software caused connection abort` and exits early.
This makes it problematic for running tests by Test Harness or internal tests automation.

#### Change overview
Make BLE connection retry on receiving such error.

#### Testing
Tested manually on Test Harness configuration - RPi 4B with chip-tool and nRF Connect platform, by executing commissioning in the loop and checking the behavior after receiving the error.

### CODE REVIEW

**Review by Damian-Nordic — APPROVED** [review]

File: src/platform/Linux/bluez/Helper.cpp [hunk 1]
```diff
@@ -1735,25 +1737,48 @@ bool BluezUnsubscribeCharacteristic(BLE_CONNECTION_OBJECT apConn)
...
 };
 
-static void ConnectDeviceDone(GObject * aObject, GAsyncResult * aResult, gpointer)
+static void ConnectDeviceDone(GObject * aObject, GAsyncResult * aResult, gpointer apParams)
 {
-    BluezDevice1 * device = BLUEZ_DEVICE1(aObject);
-    GError * error        = nullptr;
-    gboolean success      = bluez_device1_call_connect_finish(device, aResult, &error);
+    BluezDevice1 * device  = BLUEZ_DEVICE1(aObject);
+    GError * error         = nullptr;
```
Comment (erjiaqing) [inline src/platform/Linux/bluez/Helper.cpp:1749]: Minor suggestion: consider `std::unique_ptr<GError, GErrorDeleter> err` (and other `G*Deleter` w/ unique_ptr? And use a unique_ptr to hold the params so we can avoid the `exit:` & `ExitNow()` in this function.

**Review by saurabhst — APPROVED** [review]

File: src/platform/Linux/bluez/Helper.cpp (hunk 1 above)
Comment (turon) [reply src/platform/Linux/bluez/Helper.cpp:1749]: Perhaps spawn a new issue from this comment to track the request?

Comment (LuDuda) [reply src/platform/Linux/bluez/Helper.cpp:1749]: Done in (https://github.com/project-chip/connectedhomeip/issues/18286)

**Pull request merged**